### PR TITLE
avatar not showing placeholder if not source

### PR DESCRIPTION
**Pull request by jehielmartinez**
https://github.com/react-native-elements/react-native-elements/issues/2592

Fixed issue by adding a condition if `source.uri` doesn't exists
Added test for coverage

### CODE REVIEW

File: src/avatar/Avatar.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ const AvatarComponent = ({
     ));
 
   // Remove placeholder styling if we're not using image
-  const hidePlaceholder = !source;
+  const hidePlaceholder = !source || !source.uri;
```
Comment (pranshuchittora) [inline src/avatar/Avatar.js:81]: I think using `&& ` instead of `||` would be a bit optimized as the engine will check the first argument and if it's false then it won't check for the second one (right of the operator).

File: src/avatar/__tests__/Avatar.js [hunk 2]
```diff
@@ -204,5 +204,19 @@ describe('Avatar Component', () => {
...
+      const component = shallow(
+        <Avatar
+          size="medium"
+          overlayContainerStyle={{ backgroundColor: 'pink' }}
+          title="MD"
+          source={{}}
+        />
+      );
+
+      expect(component.props().style.backgroundColor).toBe('transparent');
```
Comment (pranshuchittora) [inline src/avatar/__tests__/Avatar.js:218]: Can you refactor the inline styling ` hidePlaceholder && { backgroundColor: 'transparent' },` to the stylesheet and then reference that style object here.

Why?
The styling may change might cause this test to fail. A better way is to use references to the style object directly.

**Review by pranshuchittora — CHANGES_REQUESTED** [review]
Minor Changes :)

File: src/avatar/Avatar.js (hunk 1 above)
Comment (jehielmartinez) [reply src/avatar/Avatar.js:81]: Noted, but if `source` doesn't contain `uri`, the app crashes because it can't found the property `uri` to compare. That's why I set the `||` instead of `&&`. Let me know what you think.

File: src/avatar/__tests__/Avatar.js (hunk 2 above)
Comment (jehielmartinez) [reply src/avatar/__tests__/Avatar.js:218]: Sure!

File: src/avatar/Avatar.js (hunk 1 above)
Comment (pranshuchittora) [reply src/avatar/Avatar.js:81]: Hi Jehiel, as mentioned in my previous comment that all JS engines come with a small hack/shortcut for optimizing such expressions.

```js
const x = null;

if (x && x.y) {
  console.log("x is !null and x.y as well");
} else {
  console.log("x is empty");
}

```

Just run the above snippet in Browser or Node.js and it won't fail because `x.y` will never be checked.

Comment (pranshuchittora) [reply src/avatar/Avatar.js:81]: I recommend that you test the following changes that this PR brings locally as well. You can use this app -> https://github.com/react-native-elements/react-native-elements-app
and link the local project using `npm link`. :)

Comment (jehielmartinez) [reply src/avatar/Avatar.js:81]: @pranshuchittora this is what I'm getting when changing the `||` to `&&` when testing
```
Avatar Component › Placeholders › shouldn't show placeholder if not using source

    TypeError: Cannot read property 'uri' of undefined

      79 | 
      80 |   // Remove placeholder styling if we're not using image
    > 81 |   const hidePlaceholder = !source && !source.uri;
```

Any help how to fix this?

Comment (pranshuchittora) [reply src/avatar/Avatar.js:81]: Hi so as mentioned this works for expressions , so instead try this.

`  const hidePlaceholder = (source && source.uri) ;`

**Comment by jehielmartinez** [comment]
@pranshuchittora can you check it again? Thank you

File: src/avatar/__tests__/Avatar.js [hunk 3]
```diff
@@ -204,5 +204,20 @@ describe('Avatar Component', () => {
...
+          title="MD"
+          source={{}}
+        />
+      );
+      expect(
+        component.children().props().placeholderStyle.backgroundColor
+      ).toBe('transparent');
+      expect(toJson(component)).toMatchSnapshot();
+    });
   });
```
Comment (pranshuchittora) [inline src/avatar/__tests__/Avatar.js:222]: Hi @jehielmartinez this looks great. Can you pls add the test case when source.uri is undefined/.

`source = {{  uri : undefined }}`

**Review by pranshuchittora — CHANGES_REQUESTED** [review]
LGTM, just one more edge case needs to be covered in the test suite :)

File: src/avatar/__tests__/Avatar.js (hunk 3 above)
Comment (jehielmartinez) [reply src/avatar/__tests__/Avatar.js:222]: sure!

**Comment by jehielmartinez** [comment]
@pranshuchittora test for `uri=undefined` added

**Review by pranshuchittora — APPROVED** [review]
LGTM

**Comment by flyingcircle** [comment]
@jehielmartinez contact me on the RNE slack for a small token thank you for this PR.
